src/app/models/query-models/query.ts

```diff
@@ -25,6 +26,8 @@ export class Query {
   private _observationsQuery: {data: string[]};
   // The visual indicator flags the visibility of the query
   private _visible: boolean;
+  // The information about the saved data table state
+  private _dataTableState: TransmartTableState;
```

Comment:
so the models inside the constraint-models, query-models, and table-models are what we call glowing bear models, they are the models that glowing bear operates on, and they ought not to know the existence of backend-specific models such as transmart-resource models. We need to create a service that translates backend-specific models to glowing bear models